cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -91,7 +91,7 @@ func GetStaticPodSpecs(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmap
 }
 
 // CreateStaticPodFiles creates all the requested static pod files.
-func CreateStaticPodFiles(manifestDir, patchesDir string, cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.APIEndpoint, componentNames ...string) error {
+func CreateStaticPodFiles(manifestDir, patchesDir string, cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.APIEndpoint, isDryRun bool, componentNames ...string) error {
```

Comment:
"so that later when we add rootless logic for control-plane static pod creation we can ignore user and group creation if the command/phase is run in dry-run mode". Is it not too early? And it is confused to someboy that isDryRun is added into some functions but never used in it ,I think.

Reply:
The PR for rootless control plane migration became too big. So we decided to break it into 2 parts. 1 where we plumb the dry-run flag into these functions and other where we update the static pod manifests.